app/models/git_repository.rb

```diff
@@ -30,13 +30,10 @@ def update!(executor: TerminalExecutor.new(StringIO.new))
   end
 
   def commit_from_ref(git_reference)
-    description = Dir.chdir(repo_cache_dir) do
-      IO.popen(['git', 'describe', '--long', '--tags', '--all', git_reference]) do |io|
-        io.read.strip
-      end
-    end
+    cmd = "git describe --long --tags --all #{git_reference}"
```

Comment:
Hm, fair enough. It didn't occur to me it was using `popen` to handle the shell escaping. I was just trying to simplify the code.  Will revert.